ui/src/shared/actions/flags.ts

```diff
@@ -28,3 +42,23 @@ export const setOverride = (flag: string, value: string | boolean) =>
       [flag]: value,
     },
   } as const)
+
+export const getFlags = () => async (
+  dispatch: Dispatch<Actions>
+): Promise<FlagMap> => {
+  try {
+    dispatch(setFlags(RemoteDataState.Loading))
+    const resp = await getFlagsRequest({})
+
+    if (resp.status !== 200) {
+      throw new Error(resp.data.message)
+    }
+
+    dispatch(setFlags(RemoteDataState.Done, resp.data))
+
+    return resp.data
+  } catch (error) {
+    console.error(error)
+    dispatch(setFlags(RemoteDataState.Error, null))
```

Comment:
Assuming that feature flags are internal-facing, should we also include a notification here to let the user know something went wrong to reduce confusion / investigation if something does happen?

Reply:
i dont understand

Reply:
Basically, if there’s an error in setting the feature flag, shouldn’t the user know it by receiving a notification rather than by having to investigate in some other channel if their expectation is that it should be on?

Reply:
offlined and both agreed that this is cabbage's job